tests/test_http_app/test_routes.py

```diff
@@ -0,0 +1,79 @@
+from unittest.mock import patch
+
+import pytest
+from fastapi.testclient import TestClient
+
+from imaginairy.http_app.app import app
+
+client = TestClient(app)
+
+
+@pytest.mark.asyncio()
+async def test_generate_endpoint():
+    with patch(
+        "imaginairy.http_app.stablestudio.routes.generate_image_b64"
+    ) as mock_generate_image_b64:
+        mock_generate_image_b64.return_value = base64_image_byte
+
+        test_input = {
+            "input": {
+                "prompts": [{"text": "A dog"}],
+                "sampler": {"id": "ddim"},
+                "height": 512,
+                "width": 512,
+            },
+        }
+
+        response = client.post("/api/stablestudio/generate", json=test_input)
+
+        assert response.status_code == 200
+        data = response.json()
+        assert "images" in data
+        for image in data["images"]:
+            assert image["blob"][:8] == base64_image_byte[:8].decode("utf-8")
+            assert "id" in image
+
+
+@pytest.mark.asyncio()
+async def test_list_samplers():
+    response = client.get("/api/stablestudio/samplers")
+    assert response.status_code == 200
+    assert response.json() == [
+        {"id": "ddim", "name": "ddim"},
+        {"id": "dpmpp", "name": "dpmpp"},
+    ]
+
+
+@pytest.mark.asyncio()
+async def test_list_models():
+    response = client.get("/api/stablestudio/models")
+    assert response.status_code == 200
+    assert response.json() == [
```

Comment:
this will be annoying to maintain. how about we just assert the list of `ids`

Reply:
good idea. done.